src/nft/hooks/usePermit2Approval.ts

```diff
@@ -7,16 +7,24 @@ import usePermit2Allowance, { AllowanceState } from 'hooks/usePermit2Allowance'
 import { useCallback, useMemo, useState } from 'react'
 import invariant from 'tiny-invariant'
 
+// TODO: This should be removed when the sdk is updated to include the new UR address
+const NFT_UNIVERSAL_ROUTER_MAINNET_ADDRESS = '0x4c60051384bd2d3c01bfc845cf5f4b44bcbe9de5'
+
 export default function usePermit2Approval(
   amount?: CurrencyAmount<Token>,
   maximumAmount?: CurrencyAmount<Token>,
-  enabled?: boolean
+  enabled?: boolean,
+  useNftRouter?: boolean
 ) {
   const { chainId } = useWeb3React()
 
   const allowance = usePermit2Allowance(
     enabled ? maximumAmount ?? (amount?.currency.isToken ? (amount as CurrencyAmount<Token>) : undefined) : undefined,
-    enabled && chainId ? UNIVERSAL_ROUTER_ADDRESS(chainId) : undefined
+    enabled && chainId
+      ? useNftRouter
+        ? NFT_UNIVERSAL_ROUTER_MAINNET_ADDRESS
```

Comment:
wondering how this would work when you're on a different chain rather than mainnet.. or is the NFT flow only on mainnet rn?

Reply:
We only offer NFTs on mainnet

Reply:
i would recommend handling the non mainnet + useNftRouter case in some way to prevent users from being able to call this flow if they're not on mainnet, few suggestions:

1) only pass in `useNftRouter` if user is on mainnet
2) throw Error in `usePermit2Approval` if user gets in a state with non mainnet + useNftRouter
3) default to looking up `UNIVERSAL_ROUTER_ADDRESSES` if not mainnet
4) pass in `undefined` spender if `useNftRouter` and not mainnet

Reply:
For context this cannot be called if the user is not on mainnet, the actual update approval can only be called by clicking the button which would default to switching chains over updating approval